meebee/app/src/main/java/kr/lion/meebee/ui/WeatherViewModelFactory.kt

```diff
@@ -0,0 +1,38 @@
+package kr.lion.meebee.ui
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import co.kr.lion.meebee.usecase.GetVillageForecastUseCase
+import kr.lion.meebee.Constant
+import kr.lion.meebee.datasource.village_forecast.LocalVillageForecastDataSource
+import kr.lion.meebee.datasource.village_forecast.RemoteVillageForecastDataSource
+import kr.lion.meebee.network.RetrofitInstance
+import kr.lion.meebee.network.service.VillageForecastService
+import kr.lion.meebee.repository.village_forecast.VillageForecastRepositoryImpl
+import kr.lion.meebee.ui.village_forecast.vm.VillageForecastViewModel
+
+val WeatherViewModelFactory = object : ViewModelProvider.Factory {
+    override fun <T : ViewModel> create(modelClass: Class<T>): T =
+
+        with(modelClass){
+            val getTouristSpotWeatherUseCase =
+                GetVillageForecastUseCase(
+                    VillageForecastRepositoryImpl(
+                        RemoteVillageForecastDataSource(
+                            RetrofitInstance.provideService(
+                                Constant.VILLAGE_FORECAST_BASE_URL,
+                                VillageForecastService::class.java
+                            )
+                        ),
+                        LocalVillageForecastDataSource()
+                    )
+                )
```

Comment:
와우 이정도면 Repository 에 대한 create() 함수를 하나 만들어두면 어떨까요 ? 
Factory 를 만들어도 괜찮고, 지금 단계에서는 간단하게는 확장함수를 어디 만들어놔도 괜찮겠어요. 

추후 DI 를 위해서라면 주입될 각 객체에 대한 inject 용 함수를 하나씩 만들어서 관리하는게 좋거든요. (방법은 다양하니 편한대로..)

```kotlin
fun VillageForecastRepository.create(): VillageForecastRepository {
  return VillageForecastRepositoryImpl(
    RemoteVillageForecastDataSource(
      RetrofitInstance.provideService(
        Constant.VILLAGE_FORECAST_BASE_URL,
        VillageForecastService::class.java,
      )
    ),
    LocalVillageForecastDataSource(),
  )
}
```

```suggestion
                GetVillageForecastUseCase(
                    VillageForecastRepository.create()
                )
```

Reply:
이 부분에 대해서 굉장히 불만이 많았는데 확장 함수 참 좋은 기능인거 같아요 😀

Reply:
Factory 함수에 대해 찾아봤는데 몰랐으면 큰일날뻔 했네요 ㅋㅋㅋ 😂
참 다양하고 유용한 방법들이 많은거 같아요 
네이밍은 from, of, valueOf 등등... 많은데 파라미터를 받을 때 사용하는 예시만 있어서
그냥 create( ) 로 정했어요 !
```
interface VillageForecastRepository {
    suspend fun getVillageForecast(request: Map<String, String>): VillageForecastResponse

    companion object {
        fun create(): VillageForecastRepository {
            return VillageForecastRepositoryImpl(
                RemoteVillageForecastDataSource(
                    RetrofitInstance.provideService(
                        VILLAGE_FORECAST_BASE_URL,
                        VillageForecastService::class.java,
                    )
                ),
                LocalVillageForecastDataSource(),
            )
        }
    }
}
```